StyleBox fake AA improvements (make anti aliasing size a float property)

Geometror opened the pull request:
By allowing decimal values for StyleBoxFlat's anti_aliasing_size property the user has more control over it while the crispness of the fake AA can be slightly increased as seen in this screenshot (partly fixes #35279):

![grafik](https://user-images.githubusercontent.com/50084500/128735659-0ea02c2a-2684-43fb-a004-c81ce148bb17.png)

Although it is by far from perfect, I think this is still an improvement to the StyleBox resource and could be enough for using rounded corners in godotengine/godot#51159 (@Calinou What do you think?). This change can be combined with the addition of an enum for different fake AA techniques as suggested by Calinou in https://github.com/godotengine/godot/issues/35279#issuecomment-891406752.

Changes:
-	Changed the anti-aliasing size to be a float property and set the new default value to 0.625 (determined through testing)
-	Adjusted slightly how the anti-aliasing gradient ring is calculated when a border is drawn to partly solve the problems mentioned in #35279. As shown in these screenshots, the extent of the border fake AA gradient ring matches the extent when just the infill is drawn.
CURRENT MASTER:
![grafik](https://user-images.githubusercontent.com/50084500/128739241-8e1c61b2-a395-44c3-bf9a-430cbede0b32.png)![grafik](https://user-images.githubusercontent.com/50084500/128739035-9bdac43f-8d66-4d23-bd4b-5061101f70e2.png)
THIS PR:
![grafik](https://user-images.githubusercontent.com/50084500/128738827-2576816c-d18d-40b8-a489-f8f3bfa742cc.png)![grafik](https://user-images.githubusercontent.com/50084500/128739347-706b0316-6301-404d-8b30-31698d677d5c.png)


-	Corrected some comments and variable names to fit the Godot codestyle (camelCase variables and COMMENTS FULLY IN CAPS)

I would suggest that you try it out yourself and compile this PR to see it in action since resized screenshots can't properly depict how it looks in the end. Feedback is always appreciated :)

Calinou commented:
I tested this on my default project theme PR rework. Here are the results: https://imgsli.com/NjQ5MTA/3/5

It's definitely an upgrade over what we have currently, and it should still work nicely on rotated StyleBoxes. Nice work :slightly_smiling_face:

pouleyKetchoupp reviewed (changes requested):
It's a great way to solve this issue! Thanks for looking into this.

There's just one regression I've found, it's that if you have a border with rounded corners and don't fill the inside part, the inside edges of the border are not anti-asliased anymore.

Before:
![image](https://user-images.githubusercontent.com/1075032/128903464-38a4f3b4-cdff-4707-8ecb-b6d4cad1b0b1.png)

With this PR:
![image](https://user-images.githubusercontent.com/1075032/128903573-974beee3-5c8d-452e-8181-b342296fcefd.png)

(it's also visible inside the corners without the rotation)

Apart from that, everything looks better and there's more flexibility to adjust AA, so it seems all good! It would be probably worth porting to 3.x as well, since the current state is not great and this PR's compatibility breaking is not that bad (it might just require re-adjusting AA size in some cases).

Geometror committed:
StyleBox fake AA improvements (aa_size float property)

Geometror commented:
@Calinou @pouleyKetchoupp Thanks for testing. 
I fixed the regression and changed some of the remaining comments to match Godot's codestyle (`// Abc.`) or removed them where they are unnecessary (primarily in the style_box header).

pouleyKetchoupp reviewed (approved):
Looks good!

akien-mga commented:
Thanks!

akien-mga commented:
This doesn't seem trivial to cherry-pick so it might warrant a dedicated PR if it's wanted in `3.x`.
